cvat/apps/engine/serializers.py

```diff
@@ -2275,6 +2275,35 @@ def _create_files(self, instance, files):
                     [files_model(data=instance, **f) for f in files[files_type]]
                 )
 
+class TaskReadListSerializer(serializers.ListSerializer):
+    def to_representation(self, data):
+        if isinstance(data, list) and data:
+            # Optimized prefetch only for the current page
+            page: list[models.Task] = data
+
+            # Annotate page objects
+            # We do it explicitly here and not in the LIST queryset to avoid
+            # doing the same DB computations twice - one time for the page retrieval
+            # and another one for the COUNT(*) request to get the total count
+            page_task_ids = set(t.id for t in page)
+            job_summary_fields = [
+                m.value for m in models.TaskQuerySet.JobSummaryFields.__members__.values()
+            ]
+            job_counts = {
+                task["id"]: task
+                for task in models.Task.objects
+                .filter(id__in=page_task_ids)
```

Comment:
I just thought of something WRT this and similar optimizations - how well does this behave when `page_size` is `all` and there are a lot of tasks?

Reply:
`page_size=all` must be allowed only in development deployments. It can crash the server because of OOM.